pkgs/development/tools/misc/catppuccin-catwalk/default.nix

```
@@ -1,41 +1,37 @@
 { lib
-, fetchPypi
-, python3
-,
+, fetchFromGitHub
+, rustPlatform
+, installShellFiles
 }:
 
-python3.pkgs.buildPythonApplication rec {
+rustPlatform.buildRustPackage rec {
   pname = "catppuccin-catwalk";
-  version = "0.4.0";
-  format = "pyproject";
+  version = "0.1.0";
 
-  src = fetchPypi {
-    inherit version;
-    pname = "catppuccin_catwalk";
-    hash = "sha256-5TAw5H3soxe9vLhfj1qs8uMr4ybrHlCj4zdsMzvPo6s=";
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "toolbox";
+    rev = "b38153e94622bab574224bb24a6ae953b3a849da";
+    hash = "sha256-zZRl6Xem41pBQmEoK24YR4TKiQ84nU5phBih2TF8G8g=";
   };
 
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace 'Pillow = "^' 'Pillow = ">='
-  '';
-
-  nativeBuildInputs = with python3.pkgs; [
-    poetry-core
-  ];
+  buildAndTestSubdir = "catwalk";
+  cargoLock.lockFile = "${src}/Cargo.lock";
```

```suggestion
  cargoHash = "sha256-KoxivYLzJEjWbxIkizrMpmVwUF7bfVxl13H774lzQRg=";
```
We can use cargoHash here.